### PR TITLE
SPLICE-2362 Don't compile ParserInterface.parseDataType on platforms using spark2.1.

### PR DESCRIPTION
**Pull request by msirek**
Fixes a compilation error on hdp2.6.1 and mapr6.0.0.  Those platforms compile against spark 2.1, so cannot use the optimization of SPLICE-2362 because one of the Spark API calls is only available in Spark 2.2 and above.

### CODE REVIEW

**Comment by msirek** [comment]
Jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SparkRelationalOperator.java [hunk 1]
```diff
@@ -66,9 +68,9 @@ public SparkRelationalOperator(int relOpKind,
     @Override
     public Column getColumnExpression(Dataset<Row> leftDF,
                                       Dataset<Row> rightDF,
-                                      ParserInterface parser) throws UnsupportedOperationException {
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SparkRelationalOperator.java]: Could you use a more descriptive name for the function? Maybe parser, parserFunction or something like that?

Comment (msirek) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SparkRelationalOperator.java]: OK.  Changing it to convertStringToDataTypeFunction.

**Pull request merged**